packages/react/src/components/Checkbox/CheckboxBase.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+
+interface CheckboxBaseProps {
+  checked?: boolean
+  onChange?: React.ChangeEventHandler
+  onClick?: React.MouseEventHandler
+  slots?: {
+    root?: any
+    input?: any
+  }
+  slotProps?: {
+    root?: any
+    input?: any
+  }
+  classes?: any
+}
+
+const Testhelper = (prop?: Function, propName?: string) => {
+  if (!prop) {
+    return {}
+  }
+  return {
+    [propName]: prop,
+  }
+}
+
+const CheckboxBase: React.FunctionComponent<CheckboxBaseProps> = props => {
+  const { checked, onChange, onClick, classes = {}, slots = {}, slotProps = {}, ...rest } = props
+  const { root: RootSlot = 'div', input: InputSlot = 'input' } = slots
+  const { root: rootClass, input: inputClass } = classes
+  const { root: rootProps, input: inputProps } = slotProps
+
+  const [isChecked, setIsChecked] = React.useState(!!checked)
+  const realIsChecked = checked === undefined ? isChecked : !!checked
```

Comment:
I'm having a really hard time figuring out why we need this instead of using the regular `isChecked`.